src/main/java/net/neoforged/neoforge/registries/datamaps/RegisterDataMapTypesEvent.java

```diff
@@ -31,10 +32,17 @@ public RegisterDataMapTypesEvent(Map<ResourceKey<Registry<?>>, Map<ResourceLocat
      * @param type the data map type to register
      * @param <T>  the type of the data map
      * @param <R>  the type of the registry
-     * @throws IllegalArgumentException if a type with the same ID has already been registered for that registry
+     * @throws IllegalArgumentException      if a type with the same ID has already been registered for that registry
+     * @throws UnsupportedOperationException if the registry is a non-synced datapack registry and the data map is synced
      */
     public <T, R> void register(DataMapType<R, T> type) {
         final var registry = type.registryKey();
+        if (DataPackRegistriesHooks.getDataPackRegistries().stream().anyMatch(data -> data.key().equals(registry))) {
```

Comment:
Not a huge fan of needing to iterate over every registry here, is it not possible to expose a `contains` method of some sort?

Reply:
The list isn't a `List<ResourceKey>`, it's a list of a record that also holds the codec, contains wouldn't work. I don't think that it's going to be an issue considering that there aren't thousands of dp regs.